src/verifying.rs

```diff
@@ -166,6 +167,22 @@ impl VerifyingKey {
         Scalar::from_hash(h)
     }
 
+    // Helper function for verification. Computes the _expected_ R component of the signature. The
+    // caller compares this to the real R component.  If `context.is_some()`, this does the
+    // prehashed variant of the computation using its contents.
+    #[allow(non_snake_case)]
+    fn recompute_r(
+        &self,
+        context: Option<&[u8]>,
+        signature: &InternalSignature,
+        M: &[u8],
+    ) -> EdwardsPoint {
```

Comment:
Note: this will likely return a `CompressedEdwardsY` in the process of fixing #267 . Documenting the comment here:
```
Note that this returns the compressed form of R and the caller does a byte comparison.
This means that all our verification functions do not accept non-canonically encoded R values.
See the validation criteria blog post for more details:
    https://hdevalence.ca/blog/2020-10-04-its-25519am
```